src/prettyprint.rs

```diff
@@ -1,10 +1,398 @@
+                        printer.print_str(&" = ");
+                        // no parenthesizes for expressions in declaration initializations
+                        do_prettyprint(&NodeKind::from(&assign.data), printer);
+                    }
+                    printer.print_str(&";");
+                }
+            };
+        }
+        Expr(expr) => do_prettyprint_expr(expr, printer),
+        BinaryOp(bin_op) => {
+            printer.print_str(&" ");
+            use crate::ast::BinaryOp::*;
+            let _ = match bin_op {
+                Equals => printer.print_str(&"=="),
+                NotEquals => printer.print_str(&"!="),
+                LessThan => printer.print_str(&"<"),
+                GreaterThan => printer.print_str(&">"),
+                LessEquals => printer.print_str(&"<="),
+                GreaterEquals => printer.print_str(&">="),
+                LogicalOr => printer.print_str(&"||"),
+                LogicalAnd => printer.print_str(&"&&"),
+                Add => printer.print_str(&"+"),
+                Sub => printer.print_str(&"-"),
+                Mul => printer.print_str(&"*"),
+                Div => printer.print_str(&"/"),
+                Mod => printer.print_str(&"%"),
+            };
+            printer.print_str(&" ");
+        }
+        UnaryOp(unary_op) => {
+            use crate::ast::UnaryOp::*;
+            let _ = match unary_op {
+                Not => printer.print_str(&"!"),
+                Neg => printer.print_str(&"-"),
+            };
+        }
+        PostfixOp(postfix_op) => {
+            use crate::ast::PostfixOp::*;
+            match postfix_op {
+                MethodInvocation(name, args) => {
+                    printer.print(format_args!(".{}(", name));
+                    print_argument_list(&args.data, printer);
+                    printer.print_str(&")");
+                }
+                FieldAccess(name) => {
+                    printer.print(format_args!(".{}", name));
+                }
+                ArrayAccess(expr) => {
+                    // no parenthesizes for array index expressions
+                    printer.print_str(&"[");
+                    do_prettyprint_expr(&expr.data, printer);
+                    printer.print_str(&"]");
+                }
+            }
+        }
+    }
+}
+
+fn compare_class_member(a: &'_ &ast::ClassMember<'_>, b: &'_ &ast::ClassMember) -> std::cmp::Ordering {
+    use crate::ast::ClassMemberKindDiscriminants::*;
+    let akind = ast::ClassMemberKindDiscriminants::from(&a.kind);
+    let bkind = ast::ClassMemberKindDiscriminants::from(&b.kind);
+    match (akind, bkind) {
+        (Field, Field)
+        | (Method, Method)
+        | (MainMethod, MainMethod)
+        | (Method, MainMethod)
+        | (MainMethod, Method) => a.name.cmp(&b.name),
+        (Method, Field) | (MainMethod, Field) => std::cmp::Ordering::Less,
+        (Field, Method) | (Field, MainMethod) => std::cmp::Ordering::Greater,
+    }
```

Comment:
```rust
    match (akind, bkind) {
        (Method, Field) | (MainMethod, Field) => std::cmp::Ordering::Less,
        (Field, Method) | (Field, MainMethod) => std::cmp::Ordering::Greater,
        (_, _) => a.name.cmp(&b.name),
    }
```

Reply:
Always be explicit if possible!

Reply:
Seems more like a debate. I am pro explicit.
Marking as resolved.